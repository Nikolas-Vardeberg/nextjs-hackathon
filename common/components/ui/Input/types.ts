import { HTMLAttributes } from "react";

export type InputProps = HTMLAttributes<HTMLInputElement> & {
    className?: string;
}