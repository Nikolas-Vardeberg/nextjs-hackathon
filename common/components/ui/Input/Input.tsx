import { cn } from "@/lib/cn";
import { InputProps } from "./types";

export default function Input({ className, ...props }: InputProps) {
  return (
    <input
      data-testid="input"
      className={cn(
        "flex h-9 w-full rounded-md border border-gray-300 bg-transparent px-3 py-1 text-base shadow-xs transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-black disabled:cursor-not-allowed disabled:opacity-50 md:text-sm",
        className,
      )}
      {...props}
    />
  );
}
