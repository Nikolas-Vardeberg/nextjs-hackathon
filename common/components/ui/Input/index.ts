import Input from "./Input";
import { InputProps } from "./types";

export { Input };
export type { InputProps };
