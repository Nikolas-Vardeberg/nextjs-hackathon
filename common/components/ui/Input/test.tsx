import Input from "./index";
import { render } from "@testing-library/react";

describe("Input", () => {
  it("renders the input", () => {
    const { getByTestId } = render(<Input />);
    expect(getByTestId("input")).toMatchSnapshot();
  });
});
