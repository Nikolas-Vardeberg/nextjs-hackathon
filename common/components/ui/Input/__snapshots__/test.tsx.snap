// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders the input 1`] = `
<input
  class="flex h-9 w-full rounded-md border border-gray-300 bg-transparent px-3 py-1 text-base shadow-xs transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-black disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
  data-testid="input"
/>
`;
